### PR TITLE
QGIS Vector selection algs table for parameters - 3.4

### PR DESCRIPTION
**Pull request by havatv**
Goal:

Ticket(s): #4429

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 1]
```diff
@@ -159,27 +311,67 @@ value to define the total number of features in the subset.
...
+       
+       Default: ``[Create temporary layer]``
+     - Specify the output vector layer for the randomly
+       selected features.
+       One of:
+       
+       * Create Temporary Layer (``TEMPORARY_OUTPUT``)
+       * Save to File...
+       * Save to Geopackage...
+       * Save to PostGIS Table
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
       * Save to PostGIS Table
       
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 2]
```diff
@@ -24,42 +24,99 @@ values of an attribute from the input layer.
...
+      
+       The file encoding can also be changed here.
+   * - **Extracted (non-matching)**
+     - ``FAIL_OUTPUT``
+     - [vector: any]
+       
+       Default: ``[Skip output]``
+     - Specify the output vector layer for with non-matching
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - Specify the output vector layer for non-matching
```
?

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 3]
```diff
@@ -24,42 +24,99 @@ values of an attribute from the input layer.
...
+       
+   * - **Value**
+       
+       Optional
+     - ``VALUE``
+     - [string]
+     - Value to be evaluated
+   * - **Extracted (attribute)**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```
?

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 4]
```diff
@@ -24,42 +24,99 @@ values of an attribute from the input layer.
...
+      
+       * Create Temporary Layer (``TEMPORARY_OUTPUT``)
+       * Save to File...
+       * Save to Geopackage...
+       * Save to PostGIS Table
+      
+       The file encoding can also be changed here.
+   * - **Extracted (non-matching)**
+     - ``FAIL_OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 5]
```diff
@@ -24,42 +24,99 @@ values of an attribute from the input layer.
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Extracted (attribute)**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 6]
```diff
@@ -24,42 +24,99 @@ values of an attribute from the input layer.
...
+     - Vector layer with matching features from the input
+       layer
+   * - **Extracted (non-matching)**
+     - ``FAIL_OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 7]
```diff
@@ -80,20 +137,74 @@ For more information about expressions see the :ref:`vector_expressions`.
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Matching features**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 8]
```diff
@@ -80,20 +137,74 @@ For more information about expressions see the :ref:`vector_expressions`.
...
+      
+       * Create Temporary Layer (``TEMPORARY_OUTPUT``)
+       * Save to File...
+       * Save to Geopackage...
+       * Save to PostGIS Table
+      
+       The file encoding can also be changed here.
+   * - **Non-matching**
+     - ``FAIL_OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 9]
```diff
@@ -80,20 +137,74 @@ For more information about expressions see the :ref:`vector_expressions`.
...
+     - ``INPUT``
+     - [vector: any]
+     - Input vector layer
+   * - **Expression**
+     - ``EXPRESSION``
+     - [expression]
+     - Expression to filter the vector layer
+   * - **Matching features**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 10]
```diff
@@ -80,20 +137,74 @@ For more information about expressions see the :ref:`vector_expressions`.
...
+     - Type
+     - Description
+   * - **Matching features**
+     - ``OUTPUT``
+     - [vector: any]
+     - Vector layer with matching features from the input
+       layer
+   * - **Non-matching**
+     - ``FAIL_OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 11]
```diff
@@ -112,34 +223,75 @@ spatial relationship between each feature and the features in an additional laye
...
+     - Type
+     - Description
+   * - **Extract features from**
+     - ``INPUT``
+     - [vector: any]
+     - Input vector layer
+   * - **Where the features (geometric predicate)**
+     - ``PREDICATE``
+     - [enumeration] [list]
+       Default: [0]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion

       Default: [0]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 12]
```diff
@@ -112,34 +223,75 @@ spatial relationship between each feature and the features in an additional laye
...
+       If more than one condition is chosen, at least one
+       of them (OR operation) has to be met for a feature
+       to be extracted.
+   * - **By comparing to the features from**
+     - ``INTERSECT``
+     - [vector: any]
+     - Intersection vector layer
+   * - **Extracted (location)**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 13]
```diff
@@ -112,34 +223,75 @@ spatial relationship between each feature and the features in an additional laye
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Extracted (location)**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 14]
```diff
@@ -159,27 +311,68 @@ value to define the total number of features in the subset.
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Extracted (random)**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 15]
```diff
@@ -201,30 +394,72 @@ category. Categories are defined according to a given attribute.
...
+       
+   * - **Number/percentage of selected features**
+     - ``NUMBER``
+     - [number]
+       
+       Default: 10
+     - Number or percentage of features to select
+   * - **Extracted (random stratified)**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 16]
```diff
@@ -201,30 +394,72 @@ category. Categories are defined according to a given attribute.
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Extracted (random stratified)**
+     - ``OUTPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 17]
```diff
@@ -246,21 +481,52 @@ value to define the total number of features in the subset.
...
+   :stub-columns: 0
 
-  Default: *10*
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Input layer**
+     - ``INPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 18]
```diff
@@ -290,24 +556,56 @@ No new outputs are created.
...
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 19]
```diff
@@ -328,43 +626,77 @@ No new outputs are created.
...
+   :stub-columns: 0
 
-  Default: *0*
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Input layer**
+     - ``INPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 20]
```diff
@@ -384,21 +716,52 @@ No new outputs are created.
...
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorselection.rst [hunk 21]
```diff
@@ -420,39 +783,72 @@ No new outputs are created.
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Input layer**
+     - ``INPUT``
+     - [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorselection.rst]:
```suggestion
     - [same as input]
```

**Comment by havatv** [comment]
So this means that we have decided to use [same as input].  Should that decision be reflected by / mentioned in #4502?

**Comment by DelazJ** [comment]
Based on 
@DelazJ 
> So can we go with [same as input] from now?

@havatv
> I hope so.
I have added it ([same as input]) to #4506... :-)

It was clear to me that the decision was settled ;) was it not? If not, then let me add it there too

**Comment by havatv** [comment]
You and I agreed.  Don't know about the rest of the world... ;-)

**Comment by DelazJ** [comment]
Sorry, you are right! "Rest of the world" ?
```code: rst
.. note:: It's already in use!
```

**Pull request merged**